controllers/complianceeventsapi/server.go

```diff
@@ -224,67 +249,123 @@ func splitQueryValue(value string) []string {
 // parseQueryArgs will parse the HTTP request's query arguments and convert them to a usable format for constructing
 // the SQL query. All defaults are set and any invalid query arguments result in an error being returned.
 func parseQueryArgs(queryArgs url.Values) (*queryOptions, error) {
-	parsed := &queryOptions{}
+	parsed := &queryOptions{
+		Direction:    "desc",
+		Page:         1,
+		PerPage:      20,
+		Sort:         []string{"compliance_events.timestamp"},
+		ArrayFilters: map[string][]string{},
+		Filters:      map[string][]string{},
+		NullFilters:  []string{},
+	}
 
-	perPageStr := queryArgs.Get("per_page")
+	for arg := range queryArgs {
+		valid := false
 
-	if perPageStr == "" {
-		parsed.PerPage = 20
-	} else {
-		var err error
+		for _, validQueryArg := range validQueryArgs {
+			if arg == validQueryArg {
+				valid = true
 
-		parsed.PerPage, err = strconv.ParseUint(perPageStr, 10, 64)
-		if err != nil || parsed.PerPage == 0 || parsed.PerPage > 100 {
-			return nil, fmt.Errorf("%w: per_page must be a value between 1 and 100", ErrInvalidQueryArg)
+				break
+			}
+		}
+
+		if !valid {
+			return nil, ErrInvalidQueryArg
 		}
-	}
 
-	pageStr := queryArgs.Get("page")
+		sqlName, hasSQLName := queryOptionsToSQL[arg]
 
-	if pageStr == "" {
-		parsed.Page = 1
-	} else {
-		var err error
+		value := queryArgs.Get(arg)
+		if value == "" && arg != "include_spec" {
+			// Only support null filters if it's a SQL column
+			if !hasSQLName {
+				return nil, fmt.Errorf("%w: %s must have a value", ErrInvalidQueryArgValue, arg)
+			}
+
+			parsed.NullFilters = append(parsed.NullFilters, sqlName)
 
-		parsed.Page, err = strconv.ParseUint(pageStr, 10, 64)
-		if err != nil || parsed.Page == 0 {
-			return nil, fmt.Errorf("%w: page must be a positive integer", ErrInvalidQueryArg)
+			continue
 		}
-	}
 
-	sortArgs := splitQueryValue(queryArgs.Get("sort"))
+		switch arg {
+		case "direction":
+			directionArg := queryArgs.Get("direction")
```

Comment:
I think all of the others have just used `value` instead of getting it again - was this deliberate?

Reply:
vote to Justin